src/scancode/cli.py

```diff
@@ -458,11 +465,30 @@ def scancode(ctx,
             full_root=full_root,
             ignore=user_ignore)
 
+        # Find all scans that are both enabled and have a valid function
+        # reference. This deliberately filters out the "info" scan
+        # (which always has a "None" function reference) as there is no
+        # dedicated "infos" key in the results that "plugin_only_findings.has_findings()"
+        # could check.
+        # FIXME: we should not use positional tings tuples for v[0], v[1] that are mysterious values for now
+        active_scans = [k for k, v in scanners.items() if v[0] and v[1]]
+
+        for option, post_scan_handler in plugincode.post_scan.get_post_scan_plugins().items():
+            if kwargs[option.replace('-', '_')]:
+                options['--' + option] = True
+                if not quiet:
+                    echo_stderr('Running post-scan plugin: %(option)s...' % locals(), fg='green')
+                results = post_scan_handler(active_scans, results)
+
+        # FIXME: computing len needs a list and therefore needs loading it all ahead of time
```

Comment:
This code block should be executed if and only if a post scan plugin was invoked. Otherwise in the general case this is not needed and make every scan pay for the extra memory usage. So you should first loop to get all enabled post scan plugins. Then run your loop as above on these. And only if this list is not empty, then recompute the files_count